functionary/builder/utils.py

```diff
@@ -105,6 +107,40 @@ def initiate_build(
     return build
 
 
+def _format_log_string(build_results, push_results):
+    """
+    Helper function to transform results of docker
+    push and docker build into more user friendly string.
+
+    Args:
+        build_results: itertools._tee object returned from docker build
+        push_results: generator object returned from docker push
+
+    Returns:
+        string with both build and push information in more readable format
+    """
+    build_str = ""
+    for line in build_results:
+        line_str = ""
+        for value in line.values():
+            line_str = line_str + str(value) + " "
+        build_str = build_str + line_str + "\n"
```

Comment:
It doesn't look like you need the newline here.  With it, you end up with a bunch of empty lines likes so:

```
Step 2/7 : ARG install_dir=/usr/src/app                                                                                                                                                                              
                                                                                                                                                                                                                     
                                                                                                                                                                                                                     
 ---> Using cache                                                                                                                                                                                                    
                                                                                                                                                                                                                     
 ---> a2b2693e4d1d                                                                                                                                                                                                   
                                                                                                          
Step 3/7 : COPY . $install_dir/
```